src/server/pfs/pfs.go

```diff
@@ -224,3 +274,40 @@ func IsCommitNotFinishedErr(err error) bool {
 	}
 	return commitNotFinishedRe.MatchString(err.Error())
 }
+
+// IsAmbiguousCommitErr returns true if the err is due to attempting to resolve
+// a commit without specifying a branch when it is required to uniquely identify
+// the commit.
+func IsAmbiguousCommitErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	return ambiguousCommitRe.MatchString(err.Error())
+}
+
+// IsInconsistentCommitErr returns true if the err is due to an attempt to have
+// multiple commits in a single branch within a transaction.
+func IsInconsistentCommitErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	return inconsistentCommitRe.MatchString(err.Error())
+}
+
+// IsCommitOnOutputBranchErr returns true if the err is due to an attempt to
+// start a commit on an output branch.
+func IsCommitOnOutputBranchErr(err error) bool {
+	if err == nil {
+		return false
+	}
+	return commitOnOutputBranchRe.MatchString(err.Error())
+}
+
+// IsParentNotFinishedErr returns true if the err is due to an attempt to
```

Comment:
It seems like the description of this error is different above (finishing commit on output with unfinished parent).